relayer/chain/substrate/listener.go

```diff
@@ -103,37 +94,39 @@ func (li *Listener) pollBlocks(ctx context.Context) error {
 				continue
 			}
 
-			// Get hash for latest block, sleep and retry if not ready
-			hash, err := li.conn.api.RPC.Chain.GetBlockHash(currentBlock)
+			digestItem, err := digest.GetAuxiliaryDigestItem(finalizedHeader.Digest)
```

Comment:
do we need to query for a digestitem on each block, ie, proactively, or can we just subscribe to some event like NewDigestItemCreated() and then query it reactively?

Reply:
maybe should add an issue for this?

Reply:
> do we need to query for a digestitem on each block, ie, proactively, or can we just subscribe to some event like NewDigestItemCreated() and then query it reactively? i guess it doesn't matter too much either way.

Using events as a signal wouldn't add any value and in fact would complicate things. The relayer already fetches each block header, so its much simpler and cheaper to just pro-actively look for the commitment hash in each header. 

Additionally if at all possible I'd like the relayer to avoid having to query any events, as it means we need to maintain all the event types in the relayer so that gsrpc can decode them. That caused great pain in the past for me 🙂 

As an aside, even the Polkadot-JS UI shows all digest items for a block header, which does help with debugging.

> in a future PR the relayer should query the eth contract for the latest parachain block that has been processed and then resume the substrate listener from that block so that it can catch up even if it crashes and falls behind.

Added https://github.com/Snowfork/polkadot-ethereum/issues/224

Reply:
yeh, thinking about how this integrates with the mmr stuff too - i'll draw up a flow chart now quickly and share for review.